src/test/resources/examples/githubApi/controllers/ktor/Controllers.kt

```diff
@@ -0,0 +1,823 @@
+        pullRequest: PullRequest,
+    ): ControllerResult<Unit>
+
+    /**
+     * Get a PullRequest for this Repository by ID
+     *
+     * @param parentId The unique id for the parent resource
+     * @param id The unique id for the resource
+     * @param xFlowId A custom header that will be passed onto any further requests and can be used
+     * for diagnosing.
+     *
+     * @param ifNoneMatch The RFC7232 If-None-Match header field in a request requires the server to
+     * only operate on the resource if it does not match any of the provided entity-tags. If the provided
+     * entity-tag is `*`, it is required that the resource does not exist at all.
+     *
+     */
+    public suspend fun getById(
+        call: ApplicationCall,
+        xFlowId: String?,
+        ifNoneMatch: String?,
+        parentId: String,
+        id: String,
+    ): ControllerResult<PullRequest>
+
+    /**
+     * Update the PullRequest owned by this Repository
+     *
+     * @param pullRequest
+     * @param parentId The unique id for the parent resource
+     * @param id The unique id for the resource
+     * @param ifMatch The RFC7232 If-Match header field in a request requires the server to only
+     * operate on the resource that matches at least one of the provided entity-tags. This allows clients
+     * express a precondition that prevent the method from being applied if there have been any changes
+     * to the resource.
+     *
+     * @param xFlowId A custom header that will be passed onto any further requests and can be used
+     * for diagnosing.
+     *
+     * @param idempotencyKey This unique identifier can be used to allow for clients to safely retry
+     * requests without accidentally performing the same operation twice. This is useful in cases such as
+     * network failures. This Id should remain the same for a given operation, so that the server can use
+     * it to recognise subsequent retries of the same request. Clients should be careful in using this
+     * key as subsequent requests with the same key return the same result. How the key is generated is
+     * up to the client, but it is suggested to use UUID (v4), or any other random string with enough
+     * entropy to avoid collisions.
+     *
+     */
+    public suspend fun putById(
+        call: ApplicationCall,
+        ifMatch: String,
+        xFlowId: String?,
+        idempotencyKey: String?,
+        parentId: String,
+        id: String,
+        pullRequest: PullRequest,
+    ): ControllerResult<Unit>
+
+    public companion object {
+        public fun Route.repositoriesPullRequestsRoutes(controller: RepositoriesPullRequestsController) {
+            get("/repositories/{parent-id}/pull-requests") {
+                val parentId = call.parameters.getOrFail<kotlin.String>("parent-id")
+                val xFlowId = call.request.headers["X-Flow-Id"]
+                val limit = call.request.queryParameters.getTyped<kotlin.Int>("limit")
```

Comment:
It seems that schema information is not used here, unlike it is for the Spring controllers. Wouldn't it make sense to add checks here as well?

Reply:
I assume you mean the validation constraints added as annotations to Spring and Micronaut controllers. 

Unfortunately Ktor does not have such thing built in, and we would have to either rely on an external library or implement the checks ourselves.

I have used the [RequestValidation plugin](https://ktor.io/docs/server-request-validation.html) combined Hibernate Validator with to validate the incoming body, perhaps something similar could be done for query, path and header parameters, but it would have to be custom coded.

Reply:
Yeah, that's what I meant. At the moment, the supported validations are numeric min/max, nullability, regex patterns and sequence length, all of which I think could be implemented in a few lines. But as @cjbooms said themselves below, not everything needs to be perfect from the start, so if you feel like this addition would be too much for this PR, I also won't mind if it gets merged without this feature. We should probably just create another issue and a note in the documentation in that case.

Reply:
Yes, let's add that as an issue 👍 It would be a good improvement.